cob_mimic/src/mimic_node.py

```diff
@@ -69,69 +70,100 @@
+        if self.set_mimic(goal.mimic, goal.speed, goal.repeat):
+            self._as.set_succeeded()
+        else:
+            self._as.set_aborted()
+
+    def set_mimic(self, mimic, speed, repeat):
+        rospy.loginfo("Mimic: %s", mimic)
+        file_location = '/tmp/mimic/' + mimic + '.mp4'
+        print mimic
+        if(not os.path.isfile(file_location)):
+            rospy.logerror("File not found ")
+            return False
+        # repeat cannot be 0
+        repeat = max (1, repeat)
+        self.emotion = mimic
+        self.playback_speed = speed
+        self.repeat = repeat
+        self.set_emotion()
+        return True
+
+    def main(self):
+        self.rotation = rospy.get_param('~rotation', 0)
+        # copy all videos to /tmp
+        rospy.loginfo("copying all mimic files to /tmp/mimic...")
+        file_location = roslib.packages.get_pkg_dir('cob_mimic') + '/common/*.mp4'
+        os.system("mkdir -p /tmp/mimic")
+        os.system("cp " + file_location + " /tmp/mimic")
+        rospy.loginfo("...copied all mimic files to /tmp/mimic")
+        self._ss = rospy.Service('~set_mimic', SetMimic, self.service_cb)
+        self._as = actionlib.SimpleActionServer('~set_mimic', cob_mimic.msg.SetMimicAction, execute_cb=self.action_cb, auto_start = False)
+        self._as.start()
+        self.play_defualt_emotion()
+        rospy.spin()
+
+    def setup_playback(self):
+        try:
+            if not os.path.isfile(self.default_mimic_file):
+                rospy.logerror("File not found: %s", self.default_mimic_file)
+                return
+            # run single instance of the media player
+            self.vlc_instance = vlc.Instance('--ignore-config', '--mouse-hide-timeout=0', '-q', '--no-osd', '-L', '--one-instance', '--playlist-enqueue', '--no-video-title-show')
+            self.player = self.vlc_instance.media_player_new()
+            self.player.set_fullscreen(int(True))
+        except Exception as e:
+            rospy.logerror("Something went wrong while setting up media playback: %s", str(e))
+
+    def play_defualt_emotion(self):
+        self.media = self.vlc_instance.media_new(self.default_mimic_file)
+        self.vlc_instance.vlm_set_loop(self.default_mimic_file, int(True))
+        self.media.get_mrl()
+        self.playback_speed = 1.0 # default playback speed 1.0
+        self.player.set_rate(float(self.playback_speed))
+        self.player.set_media(self.media)
+        self.player.play()
+
+    def set_emotion(self):
+        if self.repeat and self.emotion:
+            self.player.set_rate(float(self.playback_speed))
+            file_location = '/tmp/mimic/' + self.emotion + '.mp4'
+            while self.repeat:
+                self.media = self.vlc_instance.media_new(file_location)
+                self.media.get_mrl()
+                self.player.set_media(self.media)
+                self.player.play()
+                self.repeat -= 1
+                rospy.sleep(4)
```

Comment:
instead of a hardcoded sleep I'd like to have a blocking version here. should be possible to get a `playing` flag from VLC.

```
play()
while playing:
    sleep(0.1)
```


ALTERNATIVE: get the length of the video and sleep for the length of the video mulitplied by `speed`